### PR TITLE
Feat/1xxx notify chat members

### PR DESCRIPTION
**Pull request by crui3er**
Shows special notifications that inform user that another author asks for attention.
Notifications are accompanied with ring sound.
Notifications are removed only after user read till chat position when attention request was initiated.
image

### CODE REVIEW

File: src/dotnet/App.Maui/Platforms/Android/Notifications/FirebaseMessagingService.cs [hunk 1]
```diff
@@ -95,66 +80,66 @@ public override void OnMessageReceived(RemoteMessage message)
...
-        if (Log.IsEnabled(LogLevel.Debug)) {
-            var dataAsText = data.Select(c => $"'{c.Key}':'{c.Value}'").ToCommaPhrase();
-            Log.LogDebug("About to show ShowNotification '{Text}'. Data: {Data}", text, dataAsText);
+        var sentTime = new Moment(messageSentTime * 10_000).ToDateTime();
+        var title = data.Title ?? "";
+        var separatorIndex = title.IndexOf('@');
+        if (separatorIndex >= 0) {
+            // var part1 = title.Substring(0, separatorIndex).Trim();
+            var part2 = title.Substring(separatorIndex + 1).Trim();
+            title = part2;
```
Comment (frolyo) [inline src/dotnet/App.Maui/Platforms/Android/Notifications/FirebaseMessagingService.cs]: why not 
`title = title.Substring(separatorIndex + 1).Trim()`

File: src/dotnet/App.Maui/Platforms/Android/Notifications/FirebaseMessagingService.cs [hunk 2]
```diff
@@ -95,66 +80,66 @@ public override void OnMessageReceived(RemoteMessage message)
...
         }
-
-        if (Log.IsEnabled(LogLevel.Debug)) {
-            var dataAsText = data.Select(c => $"'{c.Key}':'{c.Value}'").ToCommaPhrase();
-            Log.LogDebug("About to show ShowNotification '{Text}'. Data: {Data}", text, dataAsText);
+        var sentTime = new Moment(messageSentTime * 10_000).ToDateTime();
+        var title = data.Title ?? "";
+        var separatorIndex = title.IndexOf('@');
+        if (separatorIndex >= 0) {
+            // var part1 = title.Substring(0, separatorIndex).Trim();
```
Comment (frolyo) [inline src/dotnet/App.Maui/Platforms/Android/Notifications/FirebaseMessagingService.cs]: remove

File: src/dotnet/App.Maui/Platforms/Android/Notifications/FirebaseMessagingService.cs [hunk 3]
```diff
@@ -95,66 +80,66 @@ public override void OnMessageReceived(RemoteMessage message)
...
         }
 
+        var request = new ChatAttentionRequest(chatId, data.LastEntryLocalId, sentTime, title, data.Body ?? "", data.ImageUrl ?? "");
+        ChatAttentionService.Instance.Ask(request);
+        return true;
+    }
+
+    private void ShowChatMessageNotification(NotificationData data)
+    {
+        // var intent = new Intent(this, typeof(MainActivity));
```
Comment (frolyo) [inline src/dotnet/App.Maui/Platforms/Android/Notifications/FirebaseMessagingService.cs]: is this code still needed?

File: src/dotnet/App.Maui/Platforms/Android/Notifications/NotificationData.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    public string? Body {
+        get {
+            _data.TryGetValue(Constants.Notification.MessageDataKeys.Body, out var body);
+            return body;
+        }
+    }
+
+    public string? ImageUrl {
+        get {
+            _data.TryGetValue(Constants.Notification.MessageDataKeys.ImageUrl, out var imageUrl);
```
Comment (frolyo) [inline src/dotnet/App.Maui/Platforms/Android/Notifications/NotificationData.cs]: maybe use something like GetValueOrDefault?

**Review by frolyo — APPROVED** [review]

**Review by alexis-kochetov — APPROVED** [review]

File: src/dotnet/App.Maui/Platforms/Android/Notifications/FirebaseMessagingService.cs (hunk 2 above)
Comment (crui3er) [reply src/dotnet/App.Maui/Platforms/Android/Notifications/FirebaseMessagingService.cs]: done

File: src/dotnet/App.Maui/Platforms/Android/Notifications/FirebaseMessagingService.cs (hunk 1 above)
Comment (crui3er) [reply src/dotnet/App.Maui/Platforms/Android/Notifications/FirebaseMessagingService.cs]: done

**Pull request merged**